guidelines.rst

```diff
@@ -0,0 +1,183 @@
+
+    For background and discussions, see:
+
+    - https://github.com/capi-workgroup/api-evolution/issues/26
+
+
+Meta-guidelines
+===============
+
+Don't panic!
+------------
+
+We do not expect you, fellow CPython contributor, to read and remember
+all of the guidelines.
+If you have any doubts, or if you find that the guidelines are unclear,
+incomplete or contradictory, ask the C API working group for help
+and clarifications.
+
+
+Working group approvals
+-----------------------
+
+In several cases, you should seek approval from the C API working group
+before changing public C API.
+Most notably:
+
+* when the guidelines tell you to do so,
+* when adding an exception to the rules,
+* when adding to the limited API,
+* when adding something that is likely to establish precedent,
+* when adding many APIs at once, or
+* when the guidelines are unclear or don't make sense.
+
+Approvals allow the working group to coordinate designs from multiple people,
+and to collect feedback needed to refine these guidelines.
+
+Request an approval by opening `an issue in the decisions repository`_.
+
+.. _an issue in the decisions repository: https://github.com/capi-workgroup/decisions/issues
+
+
+Exceptions to the guidelines
+----------------------------
+
+Where possible, exceptions to these guidelines should be added as
+*alternatives* to API that follows the guidelines.
+Such exceptions should be clearly identifiable from their names.
+
+If you want to add an exception that is not mentioned in the guidelines,
+please request approval from the C API workgroup.
+
+.. note:: 
+
+    For background and discussions, see:
+
+    - https://github.com/capi-workgroup/api-evolution/issues/4
+
+
+Applicability
+=============
+
+These guidelines only apply to API that is:
+
+* *Newly added*. Existing API does not necessarily follow these guidelines;
+  this PEP is not the place for plans to replace, deprecate or remove it.
+* *Public*: meant for users. That is, not in the *private* or *internal*
+  tiers as defined in the next section.
+
+Note that we have a style guide, :pep:`7`, which applies to *all* C code
+in CPython, including the API.
+
+
+API tiers
+---------
+
+Python has several tiers of C API.
+The more stable the tier is, the stricter the rules for it are:
+
+*  For **internal** API (only accessible with ``Py_BUILD_CORE``, or not declared
+   in a public header at all), you are free to ignore these guidelines entirely.
+
+*  **Private** API (not intended for users) should be *internal* if possible.
+   Otherwise it must use the ``_Py`` prefix (see :ref:`naming`).
+   The rest of these guidelines don't apply to it.
+
+*  In **unstable** API (prefixed with ``PyUnstable_``), it is OK to ignore
+   these guidelines if there is a reason to do so -- usually for performance.
+   Please document the reason in a source comment.
+
+*  In the **general** public API, these guidelines apply in full force.
+   The C API working group can grant exceptions.
```

Comment:
It would be nice to find a good term here. Unstable API is also public.

Reply:
The "Public API" is what we're currently using both in the docs and in the devguide.

Reply:
The Limited API, CPython API, and ...un-limited(?) API are all also public, so there's probably a definition required somewhere to define the scope. I don't see a problem bundling them all together for this statement (though it may read more friendly if we say something like "assume that all guidelines apply, and seek advice from the C API WG if an exception may be necessary or beneficial")

Reply:
Devguide uses *The “general” public C API* in its [list of the four tiers](https://devguide.python.org/developer-workflow/c-api/index.html).
[Docs](https://docs.python.org/3/c-api/stable.html) explain this “default” tier, and then carve out the unstable & limited tiers with different stability expectations.

(I wrote both of those; this is the best wording I found.)